admin-dev/themes/new-theme/js/pages/module/controller.js

```diff
@@ -0,0 +1,1099 @@
+
+    if ($(bulkActionSelectedSelector).length <= 0) {
+      console.warn(window.translate_javascripts['Bulk Action - One module minimum']);
+      return false;
+    }
+
+    const bulkModulesTechNames = [];
+    let moduleTechName;
+    $(bulkActionSelectedSelector).each(function bulkActionSelector() {
+      moduleTechName = $(this).data('tech-name');
+      bulkModulesTechNames.push({
+        techName: moduleTechName,
+        actionMenuObj: $(this).parent().next(),
+      });
+    });
+
+    let actionMenuObj;
+    let urlActionSegment;
+    let urlElement;
+    $.each(bulkModulesTechNames, function bulkModulesLoop(index, data) {
+      actionMenuObj = data.actionMenuObj;
+      moduleTechName = data.techName;
+
+      urlActionSegment = bulkActionToUrl[requestedBulkAction];
+
+      if (typeof self.moduleCardController !== 'undefined') {
+        // We use jQuery to get the specific link for this action. If found, we send it.
+        urlElement = $(
+          self.moduleCardController.moduleActionMenuLinkSelector + urlActionSegment,
+          actionMenuObj
+        );
+
+        if (urlElement.length > 0) {
+          self.moduleCardController._requestToController(
+            urlActionSegment,
+            urlElement,
+            forceDeletion
+          );
+        } else {
+          $.growl.error({message: window.translate_javascripts['Bulk Action - Request not available for module']
+                                        .replace('[1]', urlActionSegment)
+                                        .replace('[2]', moduleTechName)});
+        }
+      }
+    });
+
+    return true;
+  }
+
+  initActionButtons() {
+    const self = this;
+    $('body').on(
+      'click',
+      self.moduleInstallBtnSelector,
+      function initializeActionButtonsClick(event) {
+        const $this = $(this);
+        const $next = $($this.next());
+        event.preventDefault();
+
+        $this.hide();
+        $next.show();
+
+        $.ajax({
+          url: $this.data('url'),
+          dataType: 'json',
+        }).done(() => {
+          $next.fadeOut();
+        });
+      }
+    );
+
+    // "Upgrade All" button handler
+    $('body').on('click', self.upgradeAllSource, (event) => {
+      event.preventDefault();
+      $(self.upgradeAllTargets).click();
+    });
+  }
+
+  initCategorySelect() {
+    const self = this;
+    const body = $('body');
+    body.on(
+      'click',
+      self.categoryItemSelector,
+      function initializeCategorySelectClick() {
+        // Get data from li DOM input
+        self.currentRefCategory = $(this).data('category-ref');
+        self.currentRefCategory = self.currentRefCategory ? String(self.currentRefCategory).toLowerCase() : null;
+        // Change dropdown label to set it to the current category's displayname
+        $(self.categorySelectorLabelSelector).text($(this).data('category-display-name'));
+        $(self.categoryResetBtnSelector).show();
+        // Do Search on categoryRef
```

Comment:
I dont understand this comment. Is it related to `self.updateModuleVisibility()` below ?

Reply:
Was the old comments from the previous code :/

Reply:
btw I will change it